src/Umbraco.Web/Editors/Binders/ContentItemBinder.cs

```diff
@@ -49,7 +49,7 @@ public bool BindModel(HttpActionContext actionContext, ModelBindingContext bindi
             {
                 foreach (var variant in model.Variants)
                 {
-                    if (variant.Culture.IsNullOrWhiteSpace())
+                    if (variant.Culture.IsNullOrWhiteSpace() && variant.Segment.IsNullOrWhiteSpace())
```

Comment:
The original code treated invariant code separately from variant code, so I updated the "is it invariant"-check to include segments. That is, it is only invariant if both the culture and segment are null/whitespace. Perhaps invariant does not have to be treated differently, then this could indeed be removed.

Reply:
From what I can tell in code, any call to context.GetCulture would also check if the underlying thing it's working with is variant or not so seems to me like we could avoid this check and just SetCulture/SetSegment regardless since if they are null that is fine. 

I'll test out these changes - i'll merge the client side stuff in temporarily but will revert it afterwords and we can deal with the client side stuff later.